docs/rules/no-array-index-key.md

```diff
@@ -0,0 +1,23 @@
+# Prevent usage of Array index in keys
+
+Warn if an element uses an Array index in its `key`.
```

Comment:
Sure, I'll expand this document a little. I intended my next sentence to provide the reasoning, but perhaps it needs to be more explicit.